### ISSUE TITLE
Getting the error 'await is a reserved word'

### ISSUE THREAD

**Issue by tusharkhatiwada** [facebook/react-native#12092]
Hi,
I'm getting the error **`await is a reserved word`**. I've the following code in my project.

```
async handleLogin() {
    const { username, password } = this.state;
    axios.post('https://api.domain.com/login', {
      username,
      password
    })
    .then(response => {
      const res = response.data;
      if (res.success) {
        try {
          await AsyncStorage.multiSet(['token', res.token], ['CompanyName', res.company_name], ['CompanyId', res.company_id]);
        } catch (e) {
          Alert.alert('Error', 'Oops! Something happened. Please try again');
        }
      } else {
        Alert.alert('Error', 'Oops! Something happened. Please try again');
      }
    })
    .catch(err => {
      throw ('Error: ', err);
    });
  }
```

**on my .babelrc**
```
{
  "presets": [
    "react-native",
    "babel-preset-stage-3"
  ]
}
```

Am I missing something?

### Additional Information

* React Native version: 0.40.0
* Platform: Android
* Operating System: MacOS

**Comment by jaunesarmiento** [comment]
@tusharkhatiwada this answer on StackOverflow might help http://stackoverflow.com/a/36532046

**Comment by tusharkhatiwada** [comment]
@jaunesarmiento Nope. That didn't help either. I've used both `async` & `await` and my syntax seems to be correct.

**Comment by rh389** [comment]
You have `async` on `handleLogin` but not on the arrow function containing `await`.

**Comment by shahen94** [comment]
@rh389 is right, you haven't async on the arrow function, this issue is not related to React-native

**Comment by QHose** [comment]
Watch out: in my case I had to put it in front off the callback `files.forEach(async function(fileName, index) {`

**Comment by Tanumay92** [comment]
This is my code. I am getting error of await is a reserved word. any solution?

    getDistance=async()=>{
    
          
         var value=await AsyncStorage.getItem('id');
        const intervalId=BackgroundTimer.setInterval(()=>{
        var pos_lat;
        var pos_lon;
        navigator.geolocation.getCurrentPosition((position)=>{
            pos_lat=parseFloat(position.coords.latitude);
            pos_lon=parseFloat(position.coords.longitude); 
            var params_1={
            id:value,
            lat:pos_lat,
            lon:pos_lon
        };
        var formData_1=new FormData();
         for (var k in params){
         formData.append(k,params[k]);
        }
        var request_1={
            method:'POST',
            headers:{
                'X-API-KEY':'api0098911sales'
            },
            body:formData_1
        }
        var url_sendloc="http://sugoifit.in/Salesman_Tracking/index.php/api/Add_Location/add_location";
       
        var response_current=await fetch(url_sendloc,request_1);
        var json_response_current=await response_current.json();

        if(json_response_current.status=='true')
           {
            ToastAndroid.showWithGravity('Location sent!',ToastAndroid.CENTER,ToastAndroid.LONG); 
           }
           else{
             ToastAndroid.showWithGravity('Location cannot be added',ToastAndroid.CENTER,ToastAndroid.LONG);
           }  
         },
        (error)=>{
            console.log(error)
        },
        {enableHighAccuracy:false, timeout:2000, maximumAge:10000}
        );
      },10000);

**Comment by Tanumay92** [comment]
This is my code below.I am getting error of await is a reserved word. Please help me on this.

    getDistance=async()=>{
    
          
         var value=await AsyncStorage.getItem('id');
        const intervalId=BackgroundTimer.setInterval(()=>{
        var pos_lat;
        var pos_lon;
        navigator.geolocation.getCurrentPosition((position)=>{
            pos_lat=parseFloat(position.coords.latitude);
            pos_lon=parseFloat(position.coords.longitude); 
            var params_1={
            id:value,
            lat:pos_lat,
            lon:pos_lon
        };
        var formData_1=new FormData();
         for (var k in params){
         formData.append(k,params[k]);
        }
        var request_1={
            method:'POST',
            headers:{
                'X-API-KEY':'xxxxxxxxxx'
            },
            body:formData_1
        }
        var url_sendloc="xxx.cm/xx/xx";
       
        var response_current=await fetch(url_sendloc,request_1);
        var json_response_current=await response_current.json();

        if(json_response_current.status=='true')
           {
            ToastAndroid.showWithGravity('Location sent!',ToastAndroid.CENTER,ToastAndroid.LONG); 
           }
           else{
             ToastAndroid.showWithGravity('Location cannot be added',ToastAndroid.CENTER,ToastAndroid.LONG);
           }  
         },
        (error)=>{
            console.log(error)
        },
        {enableHighAccuracy:false, timeout:2000, maximumAge:10000}
        );
      },10000);

**Comment by rh389** [comment]
`navigator.geolocation.getCurrentPosition((position)=>{` should be     `navigator.geolocation.getCurrentPosition(async (position)=>{`. 

`await` needs to mark the function which immediately contains the `await`, often that could be an anonymous callback, but indentation should help you spot where they need to go.

**Comment by rh389** [comment]
Also, @QHose, be aware that with your code `forEach` won't wait for any of the async operations to complete, `appID` will be thrown away, and you won't be able to catch any rejections.

You can use `Promise.all` with `async` functions if you want to execute them in parallel and keep track of the results:

```js
try {
  const appIDs = await Promise.all(files.map((fileName, index) => async () => { // Map each file to a Promise (aka async function)
    var appName = fileName.substr(0, fileName.indexOf('.'));
    var filePath = newFolder + '\\' + fileName;
    var appID = await uploadApp(filePath, getFilesizeInBytes(filePath), appName);
    return appID;
  }));
  // All of the uploads have now completed, in parallel. `appIDs` is a populated array.
} catch (e) {
  // Something failed
}
```

Or in sequence, use an old-fashioned loop to avoid the issue with synchronously-executed callbacks:
```js
try {
  const appIDs = [];
  for (var index = 0; index < files.length; index++) {
    var fileName = files[index];
    var appName = fileName.substr(0, fileName.indexOf('.'));
    var filePath = newFolder + '\\' + fileName;
    var appID = await uploadApp(filePath, getFilesizeInBytes(filePath), appName)
    appIDs.push(appID);
  };
  // All of the uploads have now completed, in sequence. `appIDs` is a populated array.
} catch (e) {
  // Something failed
}
```
(Both of these assume the containing function is `async`)
